packages/hawtio/src/plugins/runtime/Threads.tsx

```diff
@@ -0,0 +1,333 @@
+import React, { useEffect, useState } from 'react'
+import {
+  Bullseye,
+  Button,
+  Card,
+  CardBody,
+  CodeBlock,
+  CodeBlockCode,
+  Dropdown,
+  DropdownItem,
+  DropdownToggle,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateIcon,
+  FormGroup,
+  Modal,
+  ModalVariant,
+  Pagination,
+  SearchInput,
+  Toolbar,
+  ToolbarContent,
+  ToolbarFilter,
+  ToolbarGroup,
+  ToolbarItem,
+} from '@patternfly/react-core'
+
+import { TableComposable, Tbody, Td, Th, Thead, ThProps, Tr } from '@patternfly/react-table'
+import { SearchIcon } from '@patternfly/react-icons'
+import { runtimeService } from './runtime-service'
+import { objectSorter } from '@hawtiosrc/util/objects'
+import { Thread } from '@hawtiosrc/plugins/runtime/types'
+import { ThreadInfoModal } from './ThreadInfoModal'
+
+const ThreadsDumpModal: React.FunctionComponent<{
+  isOpen: boolean
+  setIsOpen: (opened: boolean) => void
+}> = ({ isOpen, setIsOpen }) => {
+  const [threadsDump, setThreadsDump] = useState<string>('')
+  useEffect(() => {
+    const readThreadDump = async () => {
+      const threadsDump = await runtimeService.dumpThreads()
+      setThreadsDump(threadsDump)
+    }
+    if (isOpen) {
+      readThreadDump()
+    }
+  }, [isOpen])
+
+  return (
+    <Modal
+      bodyAriaLabel='Thread Dump'
+      tabIndex={0}
+      isOpen={isOpen}
+      variant={ModalVariant.large}
+      title='Thread Dump'
+      onClose={() => setIsOpen(false)}
+    >
+      <CodeBlock>
+        <CodeBlockCode>{threadsDump}</CodeBlockCode>
+      </CodeBlock>
+    </Modal>
+  )
+}
+
+export const Threads: React.FunctionComponent = () => {
+  const [threads, setThreads] = useState<Thread[]>([])
+  const [filteredThreads, setFilteredThreads] = useState<Thread[]>([])
+
+  const [page, setPage] = useState(1)
+  const [perPage, setPerPage] = useState(20)
+  const [searchTerm, setSearchTerm] = useState('')
+  const [filters, setFilters] = useState<string[]>([])
+  const [attributeMenuItem, setAttributeMenuItem] = useState('Name')
+  const [sortIndex, setSortIndex] = React.useState<number>(-1)
+  const [sortDirection, setSortDirection] = React.useState<'asc' | 'desc'>('asc')
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false)
+  const [isThreadsDumpModalOpen, setIsThreadsDumpModalOpen] = useState(false)
+  const [isThreadDetailsOpen, setIsThreadDetailsOpen] = useState(false)
+  const [currentThread, setCurrentThread] = useState<Thread>()
+  const [threadConnectionMonitoring, setThreadConnectionMonitoring] = useState(false)
+
+  useEffect(() => {
+    const readThreads = async () => {
+      const threads = await runtimeService.loadThreads()
+      setThreads(threads)
+      setFilteredThreads(threads)
+      setThreadConnectionMonitoring(await runtimeService.isThreadContentionMonitoringEnabled())
+      await runtimeService.registerLoadThreadsRequest(threads => setThreads(threads))
```

Comment:
Correct me if I'm wrong, but in this case when the scheduler updates the threads, aren't the filtered threads updated with the existing filters as well?  So, I guess the updated set of threads doesn't get updated until the user hits the search button, does it?

Reply:
nice catch!  thanks!